components/user-mgt/org.wso2.carbon.um.ws.service/src/main/java/org/wso2/carbon/um/ws/service/UserStoreManagerService.java

```diff
@@ -57,6 +59,13 @@ public void addUser(String userName, String credential, String[] roleList, Claim
 
     public void setUserClaimValues(String userName, ClaimValue[] claims, String profileName)
             throws UserStoreException {
+        // only allow alphanumeric characters for profile
+        Pattern regex = Pattern.compile("^[a-zA-Z0-9_]*$");
+        Matcher matcher = regex.matcher(profileName);
+
+        if (!matcher.matches()) {
+            throw new UserStoreException("Profile name cannot contain special characters other than underscore!");
+        }
         getUserStoreManager().setUserClaimValues(userName, convertClaimValueToMap(claims),
```

Comment:
Don't do it at this level. Write a UserOperationEventListener and do this validation in carbon-identity. Also use utility methods introduced by Malithi to do the validation instead of hardcoding stuff.